### DISCUSSION TITLE
How to get the parent nameservers?

### DISCUSSION THREAD

**Discussion by paulhermanseu** [MichaCo/DnsClient.NET#183] [General]
How do I get the nameservers from a domain, not the NS records from the zone.

**Comment by MichaConrad** [comment]
You can ask for the SOA record of the domain. If the domain exists on the server you are asking, it should return the source NS server. 
Not sure if that's what you meant though ;)

**Comment by paulhermanseu** [comment]
Hi MichaCo, thanks for your response.

I want to get the nameservers of a domainname used at the registry (parent nameserver).

For example the domain: google.de uses these nameservers at the registry DENIC:
Nserver: ns1.google.com
Nserver: ns2.google.com
Nserver: ns3.google.com
Nserver: ns4.google.com

Sidenote:
In an ideal world these nameservers would also exist as NS records in the DNS zone itself.
But on a misconfigured DNS server the DNS zone values for SOA and NS could be different (and the domainname would still be resolvable).

The nameservers of the parent are leading for domainname resolution.

**Reply by MichaConrad** [reply]
I don't think there is anything else to get the nameservers of a domain, other than asking for SOA and NS records of that domain.
To get the NS servers of the parent, just remove one label from the left side.
To get root level domain servers, query ". NS"
To get top level for .com, query ".com NS" etc..
